tests/core/pyspec/eth2spec/test/merge/block_processing/test_process_execution_payload.py

```diff
@@ -199,3 +203,182 @@ def test_bad_timestamp_regular_payload(spec, state):
     execution_payload.timestamp = execution_payload.timestamp + 1
 
     yield from run_execution_payload_processing(spec, state, execution_payload, valid=False)
+
+
+@with_merge_and_later
+@spec_state_test
+def test_bad_randao_first_payload(spec, state):
+    # pre-state
+    state = build_state_with_incomplete_transition(spec, state)
+    next_slot(spec, state)
+
+    # execution payload
+    execution_payload = build_empty_execution_payload(spec, state)
+    good_randao: Bytes32 = execution_payload.random
+    bad_randao = screw_up_bytes(good_randao)
+    # still valid because randao is ignored on this stage
+    execution_payload.random = bad_randao
+
+    yield from run_execution_payload_processing(spec, state, execution_payload)
+
+
+@with_merge_and_later
+@spec_state_test
+def test_bad_randao_regular_payload(spec, state):
+    # pre-state
+    state = build_state_with_complete_transition(spec, state)
+    next_slot(spec, state)
+
+    # execution payload
+    execution_payload = build_empty_execution_payload(spec, state)
+    good_randao: Bytes32 = execution_payload.random
+    bad_randao = screw_up_bytes(good_randao)
+    execution_payload.random = bad_randao
+
+    yield from run_execution_payload_processing(spec, state, execution_payload, valid=False)
+
+
+@with_merge_and_later
+@spec_state_test
+def test_gaslimit_zero_first_payload(spec, state):
+    # pre-state
+    state = build_state_with_incomplete_transition(spec, state)
+    next_slot(spec, state)
+
+    # execution payload
+    execution_payload = build_empty_execution_payload(spec, state)
+    execution_payload.gas_limit = uint64(0)
```

Comment:
@djrtwo from my experience any logic duplication leads to bugs when we change it someday in one place and forget about another. So I'd vote for removing it if there are no security threats by letting everything up to execution layer decision

Reply:
The intention was to put all the validations to the consensus layer except for verification that require execution state accesses like transaction validation and state root matching after the execution. The rationale behind this intention is to verify as much as we can before communicating with the execution client, also, this is gonna be a requirement for the statelessness introduction in the future, none of this arguments is that strong though. For now we decided to avoid `uint256` math that is required to validate `base_fee_per_gas` and it might make sense to remove `is_valid_gas_limit` at all.

Despite the above, I like the idea of turning `is_valid_gas_limit` to `validate_gas_limit`.